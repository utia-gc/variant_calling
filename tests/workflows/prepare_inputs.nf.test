nextflow_workflow {

    name "Test Workflow PREPARE_INPUTS"
    script "workflows/prepare_inputs.nf"
    workflow "PREPARE_INPUTS"

    test("PREPARE_INPUTS succeeds when valid samplesheet, genome, and annotation are supplied.") {
        // instantiate helper objects
        def refs = new RefsEnsembl()

        when {
            params {
                publishDirReports = "${outputDir}/reports"
                publishDirData    = "${outputDir}/data"
            }
            workflow {
                """
                input[0] = file("https://github.com/utia-gc/ngs-test/raw/ngs/samplesheets/samplesheet_mix.csv")
                input[1] = file("${refs.getGenomeGzip()}")
                input[2] = file("${refs.getGTFGzip()}")
                """
            }
        }

        then {
            // test workflow status
            assert workflow.success
            assert workflow.trace.succeeded().size() == 4

            // test samplesheet parsing
            assert workflow.out.samples.size() == 4

            // test reference genome prep
            assert workflow.out.genome
            with(workflow.out.genome) {
                assert size() == 1
                assert get(0) ==~ /^.*\/genome_I.fa/
            }

            // test genome fasta indexed
            assert workflow.out.genome_index
            with(workflow.out.genome_index) {
                assert size() == 1
                assert get(0) ==~ /^.*\/genome_I.fa.fai/
            }

            // test sequence dictionary file produced
            with(workflow.out.genome_dict) {
                assert size() == 1
                assert get(0) ==~ /^.*\/genome_I\.dict/
            }

            // test reference annotations prep
            assert workflow.out.annotations
            with(workflow.out.annotations) {
                assert size() == 1
                assert get(0) ==~ /^.*\/annotations_I.gtf/
            }

            // test workflow stability
            assert snapshot(
                workflow.out.samples,
                workflow.out.genome,
                workflow.out.genome_index,
                workflow.out.annotations,
            ).match()
        }

    }

}
