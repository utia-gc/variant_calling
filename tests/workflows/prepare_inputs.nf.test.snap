{
    "PREPARE_INPUTS succeeds when valid samplesheet, genome, and annotation are supplied.": {
        "content": [
            [
                [
                    {
                        "sampleName": "SRR1066657_GSM1299413_WT_NR_A",
                        "trimStatus": "raw",
                        "readType": "single",
                        "rgFields": {
                            "ID": "SRR1066657_GSM1299413_WT_NR_A.null",
                            "SM": "SRR1066657_GSM1299413_WT_NR_A",
                            "LB": "SRR1066657_GSM1299413_WT_NR_A",
                            "PL": "ILLUMINA"
                        }
                    },
                    "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066657_GSM1299413_WT_NR_A_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz",
                    "SRR1066657_GSM1299413_WT_NR_A_Saccharomyces_cerevisiae_RNA-Seq_50000.NOFILE:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                [
                    {
                        "sampleName": "SRR1066658_GSM1299414_WT_NR_B",
                        "trimStatus": "raw",
                        "readType": "single",
                        "rgFields": {
                            "ID": "SRR1066658_GSM1299414_WT_NR_B.null",
                            "SM": "SRR1066658_GSM1299414_WT_NR_B",
                            "LB": "SRR1066658_GSM1299414_WT_NR_B",
                            "PL": "ILLUMINA"
                        }
                    },
                    "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066658_GSM1299414_WT_NR_B_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz",
                    "SRR1066658_GSM1299414_WT_NR_B_Saccharomyces_cerevisiae_RNA-Seq_50000.NOFILE:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                [
                    {
                        "sampleName": "SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2",
                        "trimStatus": "raw",
                        "readType": "paired",
                        "rgFields": {
                            "ID": "SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2.null",
                            "SM": "SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2",
                            "LB": "SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2",
                            "PL": "ILLUMINA"
                        }
                    },
                    "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2_Saccharomyces_cerevisiae_RNA-Seq_1_50000.fastq.gz",
                    "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2_Saccharomyces_cerevisiae_RNA-Seq_2_50000.fastq.gz"
                ],
                [
                    {
                        "sampleName": "SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1",
                        "trimStatus": "raw",
                        "readType": "paired",
                        "rgFields": {
                            "ID": "SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1.null",
                            "SM": "SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1",
                            "LB": "SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1",
                            "PL": "ILLUMINA"
                        }
                    },
                    "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_1_50000.fastq.gz",
                    "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_2_50000.fastq.gz"
                ]
            ],
            [
                "genome_I.fa:md5,e050100d0a9a7d06a8accf8424ad0d98"
            ],
            [
                "genome_I.fa.fai:md5,2cd76d936cbfa386b14154506c2041b2"
            ],
            [
                "annotations_I.gtf:md5,61525d86cb937e0fe5b349215c50498c"
            ]
        ],
        "timestamp": "2023-11-07T15:02:02.49630531"
    }
}